packages/opencensus-exporter-ocagent/src/ocagent.ts

```diff
@@ -0,0 +1,313 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as protoLoader from '@grpc/proto-loader';
+import {Exporter, ExporterBuffer, ExporterConfig, logger, Logger, RootSpan, SamplerBuilder} from '@opencensus/core';
+import * as tracing from '@opencensus/nodejs';
+import * as grpc from 'grpc';
+import * as os from 'os';
+
+import {adaptRootSpan, createNode} from './adapters';
+import {opencensus, TraceServiceClient, TraceServiceConfigStream, TraceServiceExportStream} from './types';
+
+/**
+ * Options for OpenCensus Agent Exporter configuration.
+ */
+export interface OCAgentExporterOptions extends ExporterConfig {
+  serviceName: string;
+  host?: string;
+  port?: number;
+  credentials?: grpc.ChannelCredentials;
+  attributes?: Record<string, string>;
+}
+
+const DEFAULT_OPTIONS: OCAgentExporterOptions = {
+  serviceName: 'Anonymous Service',
+  host: 'localhost',
+  port: 50051,
+  credentials: grpc.credentials.createInsecure(),
+  logger: logger.logger()
+};
+
+/**
+ * grpc.Client typed event names
+ */
+enum StreamEvent {
+  Data = 'data',
+  Status = 'status',
+  Error = 'error',
+  Metadata = 'metadata'
+}
+
+/**
+ * Format and send span information to the OpenCensus Agent. Also receives and
+ * applies configuration changes from the Agent.
+ */
+export class OCAgentExporter implements Exporter {
+  // Logger
+  private logger: Logger;
+
+  // Span buffer
+  private buffer: ExporterBuffer;
+
+  // Connection objects
+  private traceServiceClient: TraceServiceClient;
+  private configStream: TraceServiceConfigStream;
+  private exportStream: TraceServiceExportStream;
+
+  // Resolved configuration options
+  private _options: OCAgentExporterOptions;
+
+  // Node properties
+  private exporterVersion: string;
+  private coreVersion: string;
+  private hostName: string;
+  private processStartTimeMillis: number;
+
+  private activeLocal = false;
+
+  constructor(options: OCAgentExporterOptions) {
+    // Get the complete set of options from the defaults and inputs
+    this._options = Object.assign({}, DEFAULT_OPTIONS, options);
+    this.logger = this._options.logger;
+
+    // Create buffer
+    this.buffer = new ExporterBuffer(this, this._options);
+
+    /**
+     * Get node properties
+     */
+    this.exporterVersion = require('../../package.json').version;
+    this.coreVersion =
+        require('../../node_modules/@opencensus/core/package.json').version;
```

Comment:
This seems a little brittle -- do you know of any third-party modules that can help with this -- or, should we have the `@opencensus/core` module directly export this as a field?

Reply:
> should we have the @opencensus/core module directly export this as a field?
You're right that this would be the better approach. Let me try to get that added.

Reply:
Looking around a bit more, there is a package that can help expose `package.json` properties, but it might be overkill if we're just looking to expose `version`: https://www.npmjs.com/package/pkginfo

We can expose the version easily by just importing the package.json and exporting from there.

The problem for _this_ PR is that it will require a version bump in the core lib before we can use it here. So I propose we do the following:
* I'm going to leave the existing version code as-is for this PR.
* I'm going to open up a second PR to add the export version to the core lib.
* I'm going to open up an issue to track using the new exported core lib version in the ocagent exporter.

Let me know if that makes sense.

Reply:
Thanks, that makes sense. I do agree that `pkginfo` is overkill.

The approach sounds good.